Orange/widgets/unsupervised/owmds.py

```diff
@@ -651,18 +652,21 @@ def _setup_plot(self, new=False):
 
         data = self.data
 
-        primitive_metas = tuple(a for a in data.domain.metas if a.is_primitive())
-        keys = [k for k, a in enumerate(data.domain.metas) if a.is_primitive()]
-        data_metas = data.metas[:, keys]
-
-        attributes = self.data.domain.attributes + (self.variable_x, self.variable_y) + \
-                     primitive_metas
+        if self.data.is_sparse():
```

Comment:
Why do we need a different logic for sparse here?

Reply:
#2693